### ISSUE TITLE
create-app : Register component gives white screen

### ISSUE THREAD

**Issue by kiranpatel11** [backstage/backstage#3687]
For app created using `create-app` cli, when clicked on "register component", white screen is shown:

chrome console error : 404 - Not found for http://localhost:7000/api/catalog/entities/by-name/User/default/guest

## Expected Behavior

Register component page should appear without any error.

## Current Behavior

White screen appears

## Steps to Reproduce

1. create a fresh app using `npx @backstage/create-app`
2. build and start backend and app.....yarn install, yarn tsc, yarn build, etc
3. Home -> Create Component -> Register Existing Component -> White screen !!

chrome console error : 404 - Not found for http://localhost:7000/api/catalog/entities/by-name/User/default/guest

**Comment by kiranpatel11** [comment]
above 404 is resolved by added below entry in app-config.yaml

```
    # Backstage example groups and users
    - type: url
      target: https://github.com/backstage/backstage/blob/master/packages/catalog-model/examples/acme-corp.yaml
```

However, it still gives white screen when clicking "Register Existing Component"

**Comment by benjdlambert** [comment]
I think this is because there is no guest user entity in the system.

Looks like this shouldn't fail though on that request.

**Comment by benjdlambert** [comment]
@kiranpatel11 is there any more logs that you can see in the console when you get the white screen?

**Comment by SudoBrendan** [comment]
We were able to reproduce this.

I'm a complete React n00b, but checking out [the docs](https://backstage.io/docs/getting-started/configure-app-with-plugins), there's no reference to the [plugin](https://github.com/backstage/backstage/tree/master/plugins/catalog-import) in the [template used by create-app](https://github.com/backstage/backstage/blob/891628b3b501a72de71459bfde59a54a62a4bc70/packages/create-app/templates/default-app/packages/app/src/App.tsx) needed to serve this endpoint (like there is [here](https://github.com/backstage/backstage/blob/master/packages/app/src/App.tsx#L37) and [here](https://github.com/backstage/backstage/blob/master/packages/app/src/App.tsx#L71-L74)), yet, the `/create` endpoint still references a button `REGISTER EXISTING COMPONENT` to link to `/catalog-import`.

We did discover that you can still register components via `APIs` > `REGISTER EXISTING API`, which links to `/register-component`.

While sniffing through the codebase/PRs open, it sounds like there's an active effort to migrate away from `/register-component` and into `/catalog-import`? Did the `REGISTER EXISTING COMPONENT` button just get caught in the crossfire?

**Comment by Rugvip** [comment]
Fixed by #3850, see changes in that PR for how to update an existing app
